frontend/src/error.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <div>
+    <Html :lang="head.htmlAttrs.lang" :dir="head.htmlAttrs.dir">
+      <Head>
+        <template v-for="link in head.link" :key="link.id">
+          <Link
+            :id="link.id"
+            :rel="link.rel"
+            :href="link.href"
+            :hreflang="link.hreflang"
+          />
+        </template>
+        <template v-for="meta in head.meta" :key="meta.id">
+          <Meta
+            :id="meta.id"
+            :property="meta.property"
+            :content="meta.content"
+          />
+        </template>
+      </Head>
+      <Body>
+        <VSkipToContentButton />
+        <NuxtLayout name="default">
+          <VFourOhFour class="flex-grow" :error="error" />
+        </NuxtLayout>
+        <div id="modal"></div>
+      </Body>
+    </Html>
+  </div>
+</template>
+
+<script setup lang="ts">
```

Comment:
Most examples of script setup I see have it above the template. Here's Vue's docs first example they show you of an SFC: https://vuejs.org/guide/introduction.html#single-file-components

Is breaking that convention intentional, or just matching our existing code (and old Vue style), or will we eventually move towards it when converting things to script setup?

Reply:
This was done to match our existing code, to minimize the number of changes. It would probably increase the cognitive load when some components have template on top and others - script. We should probably convert everything to `script setup` and _then_ change the order. On the other hand, we could also move the script to the top to make it easier to distinguish between the `script setup` and `script` components at a glance. What do you think?

Reply:
Up to you! I think it'd be good to match the typical expectations someone working with Vue would have, which I suppose would be the format the Vue documentation uses (personally I find it a bit easier to reason about too, for some reason). I don't think consistency between the old options SFC and the new script setup SFC is important, or at least not _more_ important than consistency with Vue community's expectations and documentation's examples styles.

Does not need to be something addressed in this PR though, can definitely be a follow up issue, if preferred. Moreso it would be good to decide and set a specific expectation so we're all working towards the same goal.

Reply:
I'll add the notes on the script setup components order to the PR description. It seems best to create a follow up set of issues for moving the script to the top of the file, and to convert components to script setup.